GearRent/ViewControllers/DetailsViewController.m

```diff
@@ -217,14 +235,59 @@ - (void)calendar:(JTCalendarManager *)calendar didTouchDayView:(JTCalendarDayVie
     
     if([self isInDatesSelected:dayView.date]){
         [self.datesSelected removeObject:dayView.date];
-        [self setReserveButtonText];
-        [self.calendarManager reload];
-        
     } else{ // select date
         [self.datesSelected addObject:dayView.date];
-        [self setReserveButtonText];
-        [self.calendarManager reload];
     }
+    [self setReserveButtonText];
+    [self.calendarManager reload];
+}
+
+- (void)fetchDynamicPrices{
+    NSMutableArray<NSDate *> *dates = [[self.datesForDynamicPricingSet allObjects] mutableCopy];
+    self.dateRanges = [self getDynamicPriceDateRanges:dates];
+    fetchDynamicPrice(self.listing, self.dateRanges, ^(NSDictionary<NSNumber *, NSNumber *> * result, NSError * _Nonnull error) {
+        if(error == nil) {
+            NSArray *keys = result.allKeys;
+            for(NSNumber *key in keys) {
+                [self.datesToPrices setObject:[result valueForKey:key] forKey:[NSDate dateWithTimeIntervalSince1970: [key doubleValue] / 1000]];
+            }
+            [self.calendarManager reload];
+        } else{
+            NSLog(@"%@", error);
+        }
+    });
+}
+
+- (NSMutableArray *)getDynamicPriceDateRanges:(NSMutableArray<NSDate *> *)dates {
+    NSMutableArray *result =[[NSMutableArray alloc] init];
+    if(dates.count == 0){
+        return result;
+    }
+    NSSortDescriptor *sortDescriptor = [[NSSortDescriptor alloc] initWithKey:@"self" ascending:TRUE];
+    [dates sortUsingDescriptors:[NSArray arrayWithObject:sortDescriptor]];
+    NSDate *startDate = dates[0];
+    NSDate *endDate = startDate;
+    for (int i = 1; i < dates.count; i++) {
+        NSDate *currDate = (NSDate *) dates[i];
+        NSTimeInterval timeInterval = [currDate timeIntervalSinceDate:endDate];
+        NSLog(@"%f", timeInterval);
+        if(timeInterval <= 86400.0 && timeInterval >= 0){ // days are contiguous
+            endDate = currDate;
+        } else{ // dates are not contiguous
+            NSMutableArray<NSNumber *> *dateRange = [NSMutableArray<NSNumber *> new];
+            [dateRange addObject:[NSNumber numberWithDouble:startDate.timeIntervalSince1970]];
+            [dateRange addObject:[NSNumber numberWithDouble:endDate.timeIntervalSince1970]];
+            [result addObject:dateRange];
+            startDate = currDate;
+            endDate = currDate;
+        }
+    }
+    NSMutableArray<NSNumber *> *dateRange = [NSMutableArray<NSNumber *> new];
```

Comment:
@eninodel  Since you already added the date ranges between line 278-280, do we need to add the `startDate` & `endDate` in the following lines? Will it cause any duplications?

Reply:
No, because if the days are contiguous its basically fence posting.